compendium/DeclarativeServices/test/TestConfigPolicies.cpp

```diff
@@ -0,0 +1,340 @@
+
+  // Use DS runtime service to validate the component state
+  scr::dto::ComponentDescriptionDTO compDescDTO;
+  auto compConfigs =
+    GetComponentConfigs(testBundle, componentName, compDescDTO);
+  EXPECT_EQ(compConfigs.at(0).state,
+            scr::dto::ComponentState::UNSATISFIED_REFERENCE)
+    << "UNSATISFIED_REFERENCE is expected because configuration object is not "
+       "created.";
+
+  // Get a service reference to ConfigAdmin to create the component instance.
+  auto configAdminService =
+    GetInstance<cppmicroservices::service::cm::ConfigurationAdmin>();
+  ASSERT_TRUE(configAdminService) << "GetService failed for ConfigurationAdmin";
+
+  // Create configuration object and update property
+  auto configuration  = configAdminService->GetConfiguration(componentName);
+  auto configInstance = configuration->GetPid();
+
+  cppmicroservices::AnyMap props(
+    cppmicroservices::AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
+  const std::string instanceId{ "instance1" };
+  props["uniqueProp"] = instanceId;
+  configuration->Update(props);
+
+  // The Update properties sends an asynchronous request to DS to
+  // update the properties.
+  auto result = RepeatTaskUntilOrTimeout(
+    [&compDescDTO, &compConfigs, this, &testBundle, &configInstance]() {
+      compDescDTO = dsRuntimeService->GetComponentDescriptionDTO(
+        testBundle, configInstance);
+      if (compDescDTO.name != "") {
+        compConfigs =
+          this->dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+      }
+    },
+    [&compConfigs, &instanceId]() -> bool {
+      if (compConfigs.size() == 1) {
+        auto properties = compConfigs.at(0).properties;
+        auto id = properties.find("uniqueProp");
+        if (id != properties.end()) {
+          return (id->second == instanceId);
+        }
+      }
+      return false;
+    });
+
+  ASSERT_TRUE(result) << "Timed out waiting for Update Configuration"
+                         "to complete.";
+
+  // Confirm configuration object presented and component is satisfied.
+  compConfigs = dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::SATISFIED)
+    << "SATISFIED is exepected since configuration object is created.";
+
+  //Request a service reference to the new component instance. This will
+  //cause DS to construct the instance with the updated properties.
+  auto instance = GetInstance<test::CAInterface>();
+  ASSERT_TRUE(instance) << "GetService failed for CAInterface";
+
+  //Confirm component instance was created with the correct properties
+
+  auto instanceProps = instance->GetProperties();
+  auto uniqueProp    = instanceProps.find("uniqueProp");
+
+  ASSERT_TRUE(uniqueProp != instanceProps.end())
+    << "uniqueProp not found in constructed instance";
+  EXPECT_EQ(uniqueProp->second, instanceId);
+}
+
+TEST_F(tServiceComponent, testIgnoreConfigPolicy) // DS_CAI_FTC_7
```

Comment:
Not sure we're testing the right thing here. Test procedure says "Verify that component can always be satisfied and do not use the configuration object even if present." Should use immediate=true, configuration-policy="ignore". Two tests make sense. 
Test 1 - do not create a configuration object. Verify that the component instance is created with default properties. 
Test 2 - create the configuration object. Verify that the component instance is created with default properties, not the properties from the configuration object.